feat: add C implementation for `blas/base/dspr` (#6580)
stdlib-js/stdlib: lib/node_modules/@stdlib/blas/base/dspr/README.md

### Usage

```c
#include "stdlib/blas/base/dspr.h"
```

#### c_dspr( order, uplo, N, alpha, \*X, strideX, \*AP )

Performs the symmetric rank 1 operation `A = α*x*x^T + A` where `α` is a scalar, `x` is an `N` element vector, and `A` is an `N` by `N` symmetric matrix supplied in packed form.

```c
#include "stdlib/blas/base/shared.h"

double AP[] = { 1.0, 2.0, 3.0, 1.0, 2.0, 1.0 };
const double x[] = { 1.0, 2.0, 3.0 };

> c_dspr( CblasColMajor, CblasUpper, 3, 1.0, x, 1, AP );

Reviewer: ```suggestion
c_dspr( CblasRowMajor, CblasUpper, 3, 1.0, x, 1, AP );
```

Should this be row-major?

```

The function accepts the following arguments:

-   **order**: `[in] CBLAS_LAYOUT` storage layout.

> -   **uplo**: `[in] CBLAS_UPLO` specifies whether the upper or lower triangular part of the symmetric matrix `A` should be referenced.

Reviewer: ```suggestion
-   **uplo**: `[in] CBLAS_UPLO` specifies whether the upper or lower triangular part of the symmetric matrix `A` is supplied.
```
